hilti/include/rt/util.h

```diff
@@ -359,49 +360,51 @@ class OutOfRange;
 /**
  * Parses a numerical value from a character sequence into an integer.
  * `base` must be between 2 and 26.
+ * @pre The input sequence must not be empty, i.e., we require `s != e`.
+ * @pre Base must be in the inclusive range [2, 36].
  */
 template<class Iter, typename Result>
 inline Iter atoi_n(Iter s, Iter e, int base, Result* result) {
```

Comment:
To follow up on our discussion, let's leave this as is. Returning a pair with an optional result would be the only alternative I see and that feels quite cumbersome to work with.